src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/DistributedPubSubMediator.cs

```diff
@@ -521,24 +521,24 @@ IEnumerable<IActorRef> Refs()
             foreach (var r in Refs())
             {
                 if (r == null) continue;
-                r.Forward(message);
+                r.Forward(publish.Message);
                 counter++;
             }
 
-            if (counter == 0) IgnoreOrSendToDeadLetters(message);
+            if (counter == 0) IgnoreOrSendToDeadLetters(publish);
         }
 
-        private void PublishToEachGroup(string path, object message)
+        private void PublishToEachGroup(string path, Publish publish)
         {
             var prefix = path + "/";
             var lastKey = path + "0";   // '0' is the next char of '/'
 
             var groups = ExtractGroups(prefix, lastKey).GroupBy(kv => kv.Key).ToList();
-            var wrappedMessage = new SendToOneSubscriber(message);
+            var wrappedMessage = new SendToOneSubscriber(publish.Message);
```

Comment:
Forward only the payload to the subscriber still, preserving the same behavior as before.